traversal/scanner/GraphIterator.java

```diff
@@ -226,27 +225,29 @@ private void traverse(ProcedureVertex<?, ?> procedureVertex) {
     }
 
     private void success(VertexTraverser vertexTraverser) {
-        for (ProcedureVertex<?, ?> vertex : vertexTraverser.procedureVertex.dependents()) {
-            toTraverse.add(vertex);
-            vertexTraversers.get(vertex).clear();
-        }
-        for (ProcedureVertex<?, ?> vertex : vertexTraverser.implicitDependents) {
-            toTraverse.add(vertex);
-            vertexTraversers.get(vertex).clear();
-        }
+        if (vertexTraverser.procedureVertex.order() + 1 == procedure.vertexCount()) return;
+        ProcedureVertex<?, ?> next = procedure.vertex(vertexTraverser.procedureVertex.order() + 1);
+        toTraverse.add(next);
+        vertexTraversers.get(next).clear();
```

Comment:
This could definitely be more readable with some helper functions. `vertexTraverser.procedureVertex.isLast()`, `.nextVertex()` alone would do wonders.

Reply:
updated